app/license.go

```diff
@@ -245,3 +257,66 @@ func (s *Server) RequestTrialLicense(trialRequest *model.TrialLicenseRequest) *m
 
 	return nil
 }
+
+// GenerateRenewalToken returns the current active token or generate a new one if
+// the current active one has expired
+func (s *Server) GenerateRenewalToken(expiration time.Duration) (string, *model.AppError) {
+	license := s.License()
+	if license == nil {
+		// Clean renewal token if there is no license present
+		_, err := s.Store.System().PermanentDeleteByName(model.SYSTEM_LICENSE_RENEWAL_TOKEN)
+		if err != nil {
+			mlog.Error("error removing the renewal token", mlog.Err(err))
+		}
+		return "", model.NewAppError("GenerateRenewalToken", "ent.license.generate_renewal_token.no_license", nil, "", http.StatusBadRequest)
+	}
+
+	currentToken, _ := s.Store.System().GetByName(model.SYSTEM_LICENSE_RENEWAL_TOKEN)
```

Comment:
What is the purpose of storing the token? Will storing the token start counting towards the expiration time? When will we make the request to get the link with the generated token? when they click the "Renew Now" button or just to show the button in the System Console?

Reply:
> What is the purpose of storing the token? 

Storing the token ensures that only one token is available at the same time, request from security team.

> Will storing the token start counting towards the expiration time?

Yes, when the token is generated we're adding the expiration time

> When will we make the request to get the link with the generated token? when they click the "Renew Now" button or just to show the button in the System Console?

The idea is when the button is clicked and the URL is shown or  the user is redirected

Reply:
Thanks for the insights here 👍